lib: sfloat: add a module for short float type (#8875)
nrfconnect/sdk-nrf: doc/nrf/libraries/others/sfloat.rst

.. _lib_sfloat:

Short float (SFLOAT)
####################

.. contents::
   :local:
   :depth: 2

> The library provides an implementation of the SFLOAT type.

Reviewer: Library or module?

Author: I would say it is a library because it resides in the ```nrf/lib``` directory

> You can use it to encode numbers that have the fractional part (e.g. `10.2`).

Reviewer: ```suggestion
You can use it to encode numbers that have the fractional part (for example `10.2`).
```

Author: fixed

This data type is similar to the standard FLOAT type `float`,
but it is more compact and has a lower resolution.

The IEEE 11073-20601 specification defines the SFLOAT type.
This data type encodes each number in two bytes using the following format:

.. math::

   SFLOAT = 10^e * m

> where:
>   `e` denotes exponent, encoded as a 4-bit integer in two's-complement form,
>   value range: `[-8; 7]`
> 
>   `m` denotes mantissa, encoded as a 12-bit integer in two's-complement form,
>   value range: `[-2048; 2047]`

Reviewer: ```suggestion
Where:

* `e` denotes exponent, encoded as a 4-bit integer in two's-complement form,
  value range: `[-8; 7]`.
* `m` denotes mantissa, encoded as a 12-bit integer in two's-complement form,
  value range: `[-2048; 2047]`.
```

Author: fixed

.. _lib_sfloat:

Short float (SFLOAT)
####################

.. contents::
   :local:
   :depth: 2

> 

Reviewer: As in the template, we need a few words about the library before the Overview section:

"Use this first section to give a short description (1 or 2 sentences) of the library, indicating what it does and why it should be used."

Author: Rearranged the documentation and wrote some additional content to meet these criteria

.. _lib_sfloat:

Short float (SFLOAT)
####################

.. contents::
   :local:
   :depth: 2

The library provides an implementation of the SFLOAT type, which is used to encode health data
like glucose concentration or blood pressure.

Overview
********

> You can use the library to encode numbers that have the fractional part (for example `10.2`).

Reviewer: ```suggestion
You can use the library to encode numbers that have the fractional part (for example ``10.2``).
```

> This data type is similar to the standard FLOAT type `float`,

Reviewer: ```suggestion
This data type is similar to the standard FLOAT type ``float``,
```

Author: fixed

but it is more compact and has a lower resolution.

The IEEE 11073-20601-2008 specification defines the SFLOAT type.
This data type encodes each number in two bytes using the following format:

.. math::

   SFLOAT = 10^e * m

Where:

* ``e`` denotes exponent, encoded as a 4-bit integer in two's-complement form, value range ``-8 - 7``.
* ``m`` denotes mantissa, encoded as a 12-bit integer in two's-complement form, value range ``-2048 - 2047``.

The SFLOAT type typically encodes data used in health devices.
For instance, the Glucose Concentration field in the Bluetooth Continuous Glucose Monitoring
service uses the SFLOAT type.

FLOAT conversion
****************

This library provides an API to convert the FLOAT type into SFLOAT type.
The API is only compatible with the FLOAT type from the IEEE 754 specification

> that uses the `binary32` parameter from the interchange format.

Reviewer: ```suggestion
that uses the ``binary32`` parameter from the interchange format.
```

Author: fixed
